lib/order.rb

```diff
@@ -0,0 +1,47 @@
+require_relative 'text'
+
+class Order
+  include Text
+  attr_reader :orders
+  def initialize(text_client = Text)
+    @cost = 0
+    @orders = []
+    @text_client = text_client
+  end
+
+  def choose_dish(menu, dish, quantities)
+    order = menu.show.find { |item| item[:dish] == dish }
+    @orders << {dish: dish, price: order[:price], quantities: quantities}
+  end
+
+  def checkout
+    calculate
+    result = ''
+    orders.map do |order|
+      result += "#{order[:dish]}: #{order[:quantities]} * #{order[:price]} "\
+      "= #{order[:quantities] * order[:price]} | "
+    end
+    result += "The total amount is £#{@cost}"
+  end
+
+  def cancel_orders
+    @orders = []
+  end
+
+  def pay(amount, customer)
+    fail 'You are not paying the exact amount' unless @cost == amount
+    orders.each do |order|
+      order[:paid] = true
+      order[:created_at] = (Time.now).strftime("%b %e, %Y %H:%M")
```

Comment:
Rails/TimeZone: Do not use `Time.now` without zone. Use one of `Time.zone.now`, `Time.now.current`, `Time.now.in_time_zone`, `Time.now.utc`, `Time.now.getlocal`, `Time.now.iso8601`, `Time.now.jisx0301`, `Time.now.rfc3339`, `Time.now.to_i`, `Time.now.to_f` instead. (https://github.com/bbatsov/rails-style-guide#time, http://danilenko.org/2012/7/6/rails_timezones)